### ISSUE TITLE
[BUG] npm audit uses folder name of workspace package to determine issues

### ISSUE THREAD

**Issue by peterjuras** [npm/cli#3178]
### Current Behavior:

`npm audit` appears to be checking the folder names of workspace packages against vulnerable packages. This leads to false positives, when a package with a certain name has a known vulnerability, but the actual package in the monorepo / npm workspace is published under a different package name.

Example workspace setup:

package.json
packages/
-- cli
-- cli/package.json

With the root package json:
```jsonc
{
  // ...
  "workspaces": ["packages/*"]
}
```

and `cli/package.json`:
```jsonc
{
  "name": "something-different-cli",
  // ...
}
```

The following error will be displayed on `npm audit`:
```
# npm audit report

cli  <1.0.0
Arbitrary File Write - https://npmjs.com/advisories/95
fix available via `npm audit fix`
packages/cli

1 low severity vulnerability

To address all issues, run:
  npm audit fix
```

### Expected Behavior:

`npm audit` should ignore the folder name and use the package name declared in `package.json`. I'm not sure it has to check the workspace packages against vulnerable packages at all.

### Steps To Reproduce:

1. Clone the [example repository](https://github.com/peterjuras/npm-7-ws-audit-issue)
2. Run `npm install`
3. Run `npm audit`
-> The output above should be visible
4. Rename the cli folder to e.g. `something-else-cli`
5. Delete `package-lock.json`
6. Run `npm install`
-> No issue

### Environment:

- OS: Ubuntu 20.04 (WSL2)
- Node: 14.16.1
- npm: 7.11.2

**Comment by peterjuras** [comment]
This appears to be fixed with npm@7.13.0

**Comment by Tofandel** [comment]
I'm not sure this is really fixed, this seems to be an intermittent issue maybe related to an invalid cache or invalid lock file

This issue was still present in npm@10.9.0, it comes and goes

Seems like #3192 also had a similar thing that just went away, which reinforces the cache suspicion
